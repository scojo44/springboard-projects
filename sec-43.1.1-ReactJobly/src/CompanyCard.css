.CompanyCard {
  background-color: rgba(0,0,0,30%);
  border-radius: .5rem;
  margin: .75rem;
  padding: .5rem;
  list-style-type: none;
}
.CompanyCard .link {
  display: flex;
}
.CompanyCard div {
  flex: 1;
}
.CompanyCard h3 {
  margin: 0 0 .5rem 0;
}
.CompanyCard p {
  margin: 0;
}
.CompanyCard img {
  min-width: 5rem;
  min-height: 5rem;
}
.CompanyCard a:link,
.CompanyCard a:visited {
  color: white;
  text-decoration: none;
}