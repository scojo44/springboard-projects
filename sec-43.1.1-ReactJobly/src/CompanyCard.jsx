import React from 'react'
import {Link} from 'react-router-dom';
import './CompanyCard.css'

export default function CompanyCard({handle, name, description, logoURL}) {
  return (
    <li className="CompanyCard">
      <Link to={handle} className="link">
        <div>
          <h3>{name}</h3>
          <p>{description}</p>
        </div>
        <img src={logoURL} />
      </Link>
    </li>
  )
}
