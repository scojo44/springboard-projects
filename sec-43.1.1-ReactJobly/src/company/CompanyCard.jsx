import React from 'react'
import {Link} from 'react-router-dom';
import './CompanyCard.css'

export default function CompanyCard({company}) {
  const {handle, name, description, logoURL} = company;

  return (
    <li className="CompanyCard">
      <Link to={handle} className="link">
        <div>
          <h3>{name}</h3>
          <p>{description}</p>
        </div>
        <img src={logoURL} alt={name + ' Logo'} />
      </Link>
    </li>
  );
}
