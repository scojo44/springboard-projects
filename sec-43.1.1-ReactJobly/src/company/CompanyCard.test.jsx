import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import CompanyCard from './CompanyCard';

const TEST_COMPANY = {
  handle: 'test-corp',
  name: 'Test Corp.',
  description: 'Test Corp is the leading.... yada yada yada',
  logoURL: '/test-logo.png'
}

describe('CompanyCard Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <CompanyCard company={TEST_COMPANY} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <CompanyCard company={TEST_COMPANY} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Shows some details for a company', async () => {
    const {getByText, getByAltText} = render(
      <MemoryRouter>
        <CompanyCard company={TEST_COMPANY} />
      </MemoryRouter>
    );

    expect(getByText(TEST_COMPANY.name)).toBeInTheDocument();
    expect(getByText(TEST_COMPANY.description)).toBeInTheDocument();
    expect(getByAltText(TEST_COMPANY.name + ' Logo')).toBeInTheDocument();
  });
});
