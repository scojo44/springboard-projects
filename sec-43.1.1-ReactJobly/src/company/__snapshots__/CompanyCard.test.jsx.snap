// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyCard Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="CompanyCard"
  >
    <a
      class="link"
      href="/test-corp"
    >
      <div>
        <h3>
          Test Corp.
        </h3>
        <p>
          Test Corp is the leading.... yada yada yada
        </p>
      </div>
      <img
        alt="Test Corp. Logo"
        src="/test-logo.png"
      />
    </a>
  </li>
</DocumentFragment>
`;
